staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -404,8 +404,19 @@ func (u *Unstructured) SetLabels(labels map[string]string) {
 	u.setNestedMap(labels, "metadata", "labels")
 }
 
+func (u *Unstructured) GetAnnotationsSafely() (map[string]string, error) {
+	m, _, err := NestedStringMap(u.Object, "metadata", "annotations")
+	if err != nil {
+		return nil, err
+	}
+	return m, nil
+}
+
 func (u *Unstructured) GetAnnotations() map[string]string {
-	m, _, _ := NestedStringMap(u.Object, "metadata", "annotations")
+	m, _, err := NestedStringMap(u.Object, "metadata", "annotations")
```

Comment:
We could use `GetAnnotationsSafaly` here

```go
m, err := u.GetAnnotationsSafely()
if err != nil {
  utilruntime.HandlerError(err)
}

return m
```

Reply:
Fixed